bin/add_celltypes_to_sce.R

```diff
@@ -169,29 +180,30 @@ if (!is.null(opt$singler_results)) {
 
   # add note about cell type method to metadata
   metadata(sce)$celltype_methods <- c(metadata(sce)$celltype_methods, "singler")
+} else {
+  has_singler <- FALSE # use to decide if we should assign consensus labels
 }
 
 # CellAssign results -----------------------------------------------------------
 
-if (!is.null(opt$cellassign_predictions)) {
-  # check that cellassign predictions file was provided
-  if (!file.exists(opt$cellassign_predictions)) {
-    stop("Missing CellAssign predictions file")
-  }
-
-  if (is.null(opt$cellassign_ref_file)) {
-    stop("CellAssign reference filename must be provided")
-  }
+if (file.exists(opt$cellassign_predictions)) {
+  # check that cellassign reference info is provided
+  stopifnot(
+    "CellAssign reference filename must be provided" = opt$cellassign_ref_file != "",
+    "Cell type reference metadata file does not exist" = file.exists(opt$celltype_ref_metafile),
+    "Panglo ontology reference file does not exist" = file.exists(opt$panglao_ontology_ref)
+  )
 
-  if (is.null(opt$celltype_ref_metafile)) {
-    stop("Cell type reference metadata filename must be provided")
-  }
+  # read in panglao ontology reference
+  panglao_ref_df <- readr::read_tsv(opt$panglao_ontology_ref)
 
   # if the predictions file isn't emtpy read it in
   if (file.size(opt$cellassign_predictions) > 0) {
     predictions <- readr::read_tsv(opt$cellassign_predictions)
+    has_cellassign <- TRUE # use to decide if we should assign consensus labels
   } else {
     predictions <- NULL
+    has_cellassign <- FALSE # if cell assign wasn't run correctly, don't assign consensus
   }
```

Comment:
If we check the file size earlier, then we should be good here?

```suggestion
  # read in predictions file
  predictions <- readr::read_tsv(opt$cellassign_predictions)
```